src/main/fc/fc_msp.c

```diff
@@ -1635,15 +1667,45 @@ static mspResult_e mspFcProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 
 #ifdef USE_RTC6705
     case MSP_SET_VTX_CONFIG:
-        ;
-        uint16_t tmp = sbufReadU16(src);
-        if  (tmp < 40)
-            masterConfig.vtx_channel = tmp;
-        if (current_vtx_channel != masterConfig.vtx_channel) {
-            current_vtx_channel = masterConfig.vtx_channel;
-            rtc6705_soft_spi_set_channel(vtx_freq[current_vtx_channel]);
+        if (dataSize == 2) {
+            uint16_t tmp = sbufReadU16(src);
+            if  (tmp < 40)
+                masterConfig.vtx_channel = tmp;
+            if (current_vtx_channel != masterConfig.vtx_channel) {
+                current_vtx_channel = masterConfig.vtx_channel;
+                rtc6705_soft_spi_set_channel(vtx_freq[current_vtx_channel]);
+            }
+        }
+        else {
+            return MSP_RESULT_ERROR;
         }
         break;
+#elif defined(VTX_COMMON)
```

Comment:
the data format is not really compatible at all, unfortunately... the earlier uses just one number for channel/band. However, with the old interface, MSP_VTX_CONFIG is not defined (and I'll change it as you told me to keep it this way).